float32 arg_x
float32 arg_y
---
float32 ts_x
float32 ts_y